packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -210,4 +217,70 @@ export class MetadataOrchestrator {
         return 'string';
     }
   };
+
+  public eligibilityDelegate = async (
+    requests: ApexOASEligiblePayload,
+    maxRetries: number = 3
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    let response = {};
+    let attempt = 0;
+    const languageClient = languageClientUtils.getClientInstance();
+    if (languageClient) {
+      while (attempt < maxRetries) {
+        try {
+          attempt++;
+          response = await languageClient?.sendRequest('apexoas/isEligible', requests);
+          telemetryService.sendEventData('isEligibleResponseSucceeded', {
+            classNumbers: requests.payload.length.toString()
+          });
+          break;
+        } catch (error) {
+          telemetryService.sendException('isEligibleResponseFailed', `${error} after trying ${attempt} times.`);
+
+          if (attempt >= maxRetries) {
+            // fallback TBD after we understand it better
+            throw new Error(nls.localize('cannot_get_apexoaseligibility_response'));
+          }
+        }
+      }
+    }
+    return response as ApexClassOASEligibleResponses;
+  };
+
+  public validateEligibility = async (
+    sourceUri: vscode.Uri | vscode.Uri[],
+    isMethodSelected: boolean = false
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    if (Array.isArray(sourceUri)) {
```

Comment:
Why not just add support for multiples? If the plumbing is already present then hooking the command up to do multiples becomes even simpler.